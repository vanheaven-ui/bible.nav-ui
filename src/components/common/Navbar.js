import { useState } from 'react';
import '../../styles/Navbar.css';
import { NavLink, Link } from 'react-router-dom';
import logo from '../../images/logo1.png';

const Navbar = () => {
  const [click, setClick] = useState(false);

  const handleClick = () => setClick(!click);

  return (
    <nav className="navbar">
      <div className="nav-container">
        <Link
          to="/"
          className="brand nav-logo"
          onClick={handleClick}
        >
          <img src={logo} alt="logo" width="50px" height="50px" />
          Bible.nav
        </Link>
        <ul className={click ? 'nav-menu active' : 'nav-menu'}>
          <li className="nav-item">
            <NavLink
              to="/signup"
              activeClassName="active"
              className="nav-links"
              onClick={handleClick}
            >
              Signup
            </NavLink>
          </li>
          <li className="nav-item">
            <NavLink
              to="/login"
              activeClassName="active"
              className="nav-links"
              onClick={handleClick}
            >
              Login
            </NavLink>
          </li>
        </ul>
        <button className="nav-icon" type="button" onClick={handleClick}>
          <i className={click ? 'fas fa-times' : 'fas fa-bars'} aria-label="times" />
        </button>
      </div>
    </nav>
  );
};

export default Navbar;
