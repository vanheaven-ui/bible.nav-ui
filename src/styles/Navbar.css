.navbar {
  background-color: rgba(0, 0, 0, 0.5);
  height: 80px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.2rem;
  position: sticky;
  top: 0;
  z-index: 300;
  font-family: 'Roboto Condensed', sans-serif;
}

.nav-container {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  max-width: 1500px;
  margin: 0 20px;
}

.nav-logo {
  color: #ec5200;
  align-items: center;
  margin-left: 20px;
  cursor: pointer;
  text-decoration: none;
  font-size: 2rem;
  margin-right: auto;
  text-shadow: 1px 1px 2px #fff;
  border-bottom: 2px solid;
  transition: all 0.5s ease-in-out;
}

.nav-logo:hover {
  transform: scale(1.5) translateY(-2px);
  color: #fff;
  text-shadow: 1px 2px 3px #ec5200;
  border-bottom: 2px solid;
}

.nav-menu {
  display: flex;
  list-style: none;
  text-align: center;
  margin: 0;
}

.profile span small {
  color: #ec5200;
  font-weight: 900;
}

.nav-links {
  color: #fff;
  text-decoration: none;
  padding: 0.5rem 1rem;
  height: 100%;
  border-bottom: 3px solid transparent;
}

.fa-code {
  margin-left: 1rem;
}

.nav-item {
  line-height: 40px;
  margin-right: 1rem;
}

.profile {
  font-size: 22px;
  color: #fff;
}

.nav-item::after {
  content: "";
  display: block;
  height: 3px;
  width: 0;
  background: transparent;
  transition: width 0.7s ease, background-color 0.5s ease;
}

.nav-item:hover {
  color: #fff;
}

.nav-item:hover::after {
  width: 100%;
  background: #ec5200;
}

.nav-item .active {
  color: #ec5200;
  border: 1px solid #ec5200;
}

.nav-icon {
  display: none;
}

@media screen and (max-width: 768px) {
  .nav-menu {
    display: flex;
    flex-direction: column;
    width: 100%;
    border-top: 1px solid #fff;
    position: absolute;
    top: 80px;
    left: -110%;
    opacity: 1;
    transition: all 0.5s ease;
  }

  .nav-menu.active {
    background: rgba(0, 0, 0, 0.8);
    left: 0;
    opacity: 1;
    transition: all 0.5s ease;
    z-index: 300;
    padding-left: 0;
  }

  .nav-menu.active li:first-child {
    color: #ec5200;
    font-weight: 700;
  }

  .nav-item .active {
    color: #ec5200;
    border: none;
  }

  .nav-links {
    padding: 1.5rem;
    width: 100%;
    display: table;
  }

  .nav-icon {
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    transform: translate(-100%, 60%);
    font-size: 1.8rem;
    cursor: pointer;
    color: #ec5200;
    background-color: transparent;
    padding: 0 10px;
    border: none;
    transition: all 0.5s ease-out;
  }
}
