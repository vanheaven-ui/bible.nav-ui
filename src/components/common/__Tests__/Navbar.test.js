import React from 'react';
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import rootDiv from '../../../testHelper';
import '@testing-library/jest-dom';
import Navbar from '../Navbar';

it('renders without crashing', () => {
  ReactDOM.render(<Router><Navbar /></Router>, rootDiv());
});

it('does not change unexpectedly', () => {
  const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the correct wrapper element', () => {
  const { container } = render(<Router><Navbar /></Router>);
  expect(container.firstChild.classList.contains('navbar'));
});
