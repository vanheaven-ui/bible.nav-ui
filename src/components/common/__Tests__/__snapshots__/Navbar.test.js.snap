// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not change unexpectedly 1`] = `
<nav
  className="navbar"
>
  <div
    className="nav-container"
  >
    <a
      className="brand nav-logo"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        height="50px"
        src="logo1.png"
        width="50px"
      />
      Bible.nav
    </a>
    <ul
      className="nav-menu"
    >
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-links"
          href="/signup"
          onClick={[Function]}
        >
          Signup
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          aria-current={null}
          className="nav-links"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
    <button
      className="nav-icon"
      onClick={[Function]}
      type="button"
    >
      <i
        aria-label="times"
        className="fas fa-bars"
      />
    </button>
  </div>
</nav>
`;
